tests/lib/fakestore/refresh/snapAsserts.go

```diff
@@ -0,0 +1,97 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package refresh
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/systestkeys"
+)
+
+func snapNameFromPath(snapPath string) string {
+	return strings.Split(filepath.Base(snapPath), "_")[0]
+}
+
+// TODO: also support reading/copying form a store snap
+
+func NewSnapRevision(targetDir string, snap string, headers map[string]interface{}) error {
+	db, err := newAssertsDB()
+	if err != nil {
+		return err
+	}
+	digest, size, err := asserts.SnapFileSHA3_384(snap)
+	if err != nil {
+		return err
+	}
+
+	headers["authority-id"] = "testrootorg"
+	headers["snap-sha3-384"] = digest
+	if _, ok := headers["developer-id"]; !ok {
```

Comment:
I would make a for loop over a  map[string]interface{} with the fallbacks but that's me